docs/proposals/odo-v3-cli/odo-delete-component.md

```diff
@@ -0,0 +1,17 @@
+# `odo delete component`
+Delete component.
+By default it deletes component from cluster (deletes all resources from both `odo dev` and `odo deploy` that were crated on the cluster that belong to the component) that in the current directory.
+
+Similarly to `odo describe component` this commands works in two modes.
+Without `--component` `--application` or `--namespace` it works with the component in the current directory.
+If user defines `--component` and `--application` than it works with the component that should exist on the cluster, and it should not touch local component even if the current directory is component directory.
```

Comment:
> it should not touch local component even if the current directory is component directory.

Again, why not make odo mimic the behaviour of git here?

Also, what kind of commands do we expect from users here?
```sh
$ odo delete component --component someTerribleComponent --application blah
$ odo delete component --component blah --application moreBlah --namespace yikes
$ odo delete component
```

Since when did odo start using the term namespace instead of project? Can we do more research/thinking before coming up with what we want the CLI UX to look like?

Reply:
Odo needs to be able to delete component form the cluster even without having access to a devfile.yaml that was originally used to create that component.

If you have a better way how to achieve that I'm open to suggestions.

Reply:
> Odo needs to be able to delete component form the cluster even without having access to a devfile.yaml that was originally used to create that component.

OK. I won't annoy you with "did a user ask for it", because I want to be able to do "constructive criticism" instead of "ranting".



> If you have a better way how to achieve that I'm open to suggestions.

For a component "odcComponent" created by ODC in the namespace "myproject", user can delete it by doing:
```sh
$ odo project set myproject
$ odo delete component odcComponent -f
```
No use of `--component` or `--application` or `--namespace` flags there. It becomes a two-step process, but it is succinct and clear, and at code level it's simpler to maintain.

Reply:
>OK. I won't annoy you with "did a user ask for it", because I want to be able to do "constructive criticism" instead of "ranting".

yes, the same as for `odo describe` this is based on user feedback.


I've remove `--aplication` flag from all commands, for now. But `--name` and `--namespace` needs to be there.